lib/workers/repository/dashboard-package-files.ts

```diff
@@ -0,0 +1,60 @@
+import type { RenovateConfig } from '../../config/types';
+import type { PackageFile } from '../../modules/manager/types';
+
+export class DashboardPackageFiles {
+  private static data: Map<string, Record<string, PackageFile[]> | null> =
+    new Map<string, Record<string, PackageFile[]>>();
+
+  public static add(
+    baseBranch: string,
+    packageFiles: Record<string, PackageFile[]> | null
+  ): void {
+    this.data.set(baseBranch, packageFiles);
+  }
+
+  public static clear(): void {
+    this.data.clear();
+  }
+
+  public static getDetectedDependencies(config: RenovateConfig): string {
+    const title = `## Detected dependencies\n\n`;
+    const none = 'None detected\n\n';
+    const pad = this.data.size > 1; // padding condition for a multi base branch repo
+    let deps = '';
+
+    if (!config.dependencyDashboardDetectedDeps) {
+      return '';
+    }
+
+    for (const [branch, packageFiles] of this.data) {
+      deps += pad ? `<details><summary>Branch ${branch}\n</summary>\n\n` : '';
+      if (packageFiles === null) {
+        deps += none + '\n';
+        continue;
+      }
+
+      const managers = Object.keys(packageFiles);
+      if (managers.length === 0) {
+        deps += none + '\n';
+        continue;
+      }
+
+      for (const manager of managers) {
+        deps += `${
+          pad ? '<ul>' : ''
+        }<details><summary>${manager}</summary>\n\n`;
+        for (const packageFile of packageFiles[manager]) {
+          deps += `<ul><details><summary>${packageFile.packageFile}</summary>\n\n`;
+          for (const dep of packageFile.deps) {
+            deps += ` - \`${dep.depName}@${dep.currentValue}\`\n`;
```

Comment:
Is `currentValue` ever null or an empty string? If so might be better to drop the trailing `@`

Reply:
`currentValue` is defined as string | null. not sure thought if can be null/empty string in the wild, therefore removed the trailing @.

we could also just check for it directly and preserve the @.
it might be only me, but below looks kinda off:

 - `cookie-parser ^1.4.5`
 - `express ~4.17.1`
 - `express-handlebars >=5.3.4`
 - `geoip-lite 1.4.*`
 - `nodemailer 6.7.0`
 - `redis 3.1.2-3.4.0`
 - `dotenv 10.0.0`
 - `nodemon 2.0.14`